src/components/Modal/Modal.js

```diff
@@ -12,36 +10,36 @@ class Modal extends Component {
   };
 
   static defaultProps = {
-    actionRequired: false,
     ref: 'modal',
   };
 
-  componentDidMount() {
-    this.initializeModal();
-  }
-
-  initializeModal = () => {
-    this.elm = $(React.findDOMNode(this.refs[this.props.ref]));
-    this.elm.modal();
-  }
+  state = {isShown: false}
 
   showModal = () => {
-    this.elm = $(React.findDOMNode(this.refs[this.props.ref]));
-    if (this.props.actionRequired) {
-      // Semantic UI settings, do not allow modal to close without action on button click
-      this.elm.modal('setting', 'closable', false).modal('show');
-    } else {
-      this.elm.modal('show');
-    }
+    this.setState({isShown: true});
   }
 
   hideModal = () => {
-    this.elm = $(React.findDOMNode(this.refs[this.props.ref]));
-    this.elm.modal('hide');
+    this.setState({isShown: false});
   };
 
   render() {
-    let classes = classNames('sd-modal', this.props.className, 'ui', {small: this.props.small}, 'modal');
+    let classes;
+    if (this.state.isShown) {
+      classes = classNames(
+        'sd-modal',
+        this.props.className,
+        'ui',
+        {small: this.props.small},
+        'modal',
+        'transition',
+        'visible',
+        'active',
+        );
+    } else {
+      classes = classNames('sd-modal', this.props.className, 'ui', {small: this.props.small}, 'modal');
+    }
```

Comment:
Don't add Semantic classes on Stardust components, extend the Stardust component to expose props instead.

Keep DRY, let's not repeat `classNames()` and the list of classes.  Use the if conditional inside a single call to classNames to toggle classes on/off.